[BUG] [gradle] Tasks that inherit ValidateTask will fail if recommend is not assigned.

#### Bug Report Checklist

- [x] Have you provided a full/minimal spec to reproduce the issue?
- [x] Have you validated the input using an OpenAPI validator ([example](https://apidevtools.org/swagger-parser/online/))?
- [x] What's the version of OpenAPI Generator used?
- [x] Have you search for related issues/PRs?
- [x] What's the actual output vs expected output?
- [ ] [Optional] Bounty to sponsor the fix ([example](https://www.bountysource.com/issues/66123212-javascript-client-produces-a-wrong-object-for-a-string-enum-type-that-is-used-with-ref))

##### Description

I am using the openapi-generator-gradle-plugin.
I've noticed that the default value doesn't work well with ValidateTask.
I tried the [samples](https://github.com/OpenAPITools/openapi-generator/tree/master/modules/openapi-generator-gradle-plugin/samples/local-spec
) in this repository and the problem was reproduced.

Although `recommend` has a default value, tasks that inherit ValidateTask will fail if `recommend` is not assigned.
https://github.com/OpenAPITools/openapi-generator/tree/master/modules/openapi-generator-gradle-plugin#openapivalidate

```gradle
task validateGoodSpec(type: org.openapitools.generator.gradle.plugin.tasks.ValidateTask){
    inputSpec = "$rootDir/petstore-v3.0.yaml".toString()
//	recommend = true
//      recommend = false
}
```

##### openapi-generator version

I have tried with 4.3.0 and 4.3.1.

##### OpenAPI declaration file content or url

https://github.com/OpenAPITools/openapi-generator/blob/master/modules/openapi-generator-gradle-plugin/samples/local-spec/petstore-v3.0.yaml

##### Command line used for generation

`./gradlew validateGoodSpec`
`./gradlew -PopenApiGeneratorVersion=4.3.0 validateGoodSpec`
`./gradlew -PopenApiGeneratorVersion=4.3.1 validateGoodSpec`

##### Steps to reproduce

I clone this repository and exec validateGoodSpec.

`git clone git@github.com:OpenAPITools/openapi-generator.git`
`cd openapi-generator/modules/openapi-generator-gradle-plugin/samples/local-spec`
`./gradlew validateGoodSpec`

```
> Task :validateGoodSpec FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':validateGoodSpec'.
> No value has been specified for this provider.

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 895ms
1 actionable task: 1 executed
```

I have explicitly assigned true or false to `recommend`.
Then the task was successful.

```
> Task :validateGoodSpec
Validating spec /****/openapi-generator/modules/openapi-generator-gradle-plugin/samples/local-spec/petstore-v3.0.yaml
Spec is valid.

BUILD SUCCESSFUL in 1s
1 actionable task: 1 executed
```

##### Related issues/PRs

https://github.com/OpenAPITools/openapi-generator/issues/4643
https://github.com/OpenAPITools/openapi-generator/pull/5183

##### Suggest a fix

I think it should be set default value before `get` is called at https://github.com/OpenAPITools/openapi-generator/blob/5fe34fbd74e8078f7f3ee180bfb302ab2d0662d8/modules/openapi-generator-gradle-plugin/src/main/kotlin/org/openapitools/generator/gradle/plugin/tasks/ValidateTask.kt#L72

--- a/modules/openapi-generator-gradle-plugin/src/main/kotlin/org/openapitools/generator/gradle/plugin/extensions/OpenApiGeneratorValidateExtension.kt
+++ b/modules/openapi-generator-gradle-plugin/src/main/kotlin/org/openapitools/generator/gradle/plugin/extensions/OpenApiGeneratorValidateExtension.kt
@@ -33,12 +33,5 @@ open class OpenApiGeneratorValidateExtension(project: Project) {
     /**
      * Whether to offer recommendations related to the validated specification document.
      */
-    val recommend = project.objects.property<Boolean?>()
-
-    init {
-        applyDefaults()
-    }
-
-    @Suppress("MemberVisibilityCanBePrivate")
-    fun applyDefaults() = recommend.set(true)
+    val recommend = project.objects.property<Boolean>().convention(true)
 }

--- a/modules/openapi-generator-gradle-plugin/src/main/kotlin/org/openapitools/generator/gradle/plugin/tasks/ValidateTask.kt
+++ b/modules/openapi-generator-gradle-plugin/src/main/kotlin/org/openapitools/generator/gradle/plugin/tasks/ValidateTask.kt
@@ -58,7 +58,7 @@ open class ValidateTask : DefaultTask() {
 
     @Optional
     @Input
-    val recommend = project.objects.property<Boolean?>()
+    val recommend = project.objects.property<Boolean>().convention(true)
 
     @get:Internal
     @set:Option(option = "input", description = "The input specification.")

--- a/modules/openapi-generator-gradle-plugin/src/test/kotlin/ValidateTaskDslTest.kt
+++ b/modules/openapi-generator-gradle-plugin/src/test/kotlin/ValidateTaskDslTest.kt
@@ -144,4 +144,77 @@ class ValidateTaskDslTest : TestBase() {
             "Expected a failed run, but found ${result.task(":openApiValidate")?.outcome}"
         )
     }
+
+    @Test(dataProvider = "gradle_version_provider")
+    fun `validateGoodSpec as defined task should succeed on valid spec`(gradleVersion: String?) {
+        // Arrange
+        val projectFiles = mapOf(
+            "spec.yaml" to javaClass.classLoader.getResourceAsStream("specs/petstore-v3.0.yaml")
+        )
+
+        withProject(
+            """
+            | plugins {
+            |   id 'org.openapi.generator'
+            | }
+            |
+            | task validateGoodSpec(type: org.openapitools.generator.gradle.plugin.tasks.ValidateTask) {
+            |    inputSpec.set(file("spec.yaml").absolutePath)
+            | }
+        """.trimMargin(), projectFiles
+        )
+
+        // Act
+        val result = getGradleRunner(gradleVersion)
+            .withProjectDir(temp)
+            .withArguments("validateGoodSpec")
+            .withPluginClasspath()
+            .build()
+
+        // Assert
+        assertTrue(
+            result.output.contains("Spec is valid."),
+            "Unexpected/no message presented to the user for a valid spec."
+        )
+        assertEquals(
+            SUCCESS, result.task(":validateGoodSpec")?.outcome,
+            "Expected a successful run, but found ${result.task(":validateGoodSpec")?.outcome}"
+        )
+    }
+
+    @Test(dataProvider = "gradle_version_provider")
+    fun `validateBadSpec as defined task should fail on invalid spec`(gradleVersion: String?) {
+        // Arrange
+        val projectFiles = mapOf(
+            "spec.yaml" to javaClass.classLoader.getResourceAsStream("specs/petstore-v3.0-invalid.yaml")
+        )
+        withProject(
+            """
+            | plugins {
+            |   id 'org.openapi.generator'
+            | }
+            |
+            | task validateBadSpec(type: org.openapitools.generator.gradle.plugin.tasks.ValidateTask) {
+            |    inputSpec.set(file("spec.yaml").absolutePath)
+            | }
+        """.trimMargin(), projectFiles
+        )
+
+        // Act
+        val result = getGradleRunner(gradleVersion)
+            .withProjectDir(temp)
+            .withArguments("validateBadSpec")
+            .withPluginClasspath()
+            .buildAndFail()
+
+        // Assert
+        assertTrue(
+            result.output.contains("Spec is invalid."),
+            "Unexpected/no message presented to the user for an invalid spec."
+        )
+        assertEquals(
+            FAILED, result.task(":validateBadSpec")?.outcome,
+            "Expected a failed run, but found ${result.task(":validateBadSpec")?.outcome}"
+        )
+    }
 }

--- a/modules/openapi-generator-gradle-plugin/src/test/kotlin/ValidateTaskDslTest.kt
+++ b/modules/openapi-generator-gradle-plugin/src/test/kotlin/ValidateTaskDslTest.kt
@@ -217,4 +217,88 @@ class ValidateTaskDslTest : TestBase() {
             "Expected a failed run, but found ${result.task(":validateBadSpec")?.outcome}"
         )
     }
+
+    @Test(dataProvider = "gradle_version_provider")
+    fun `openApiValidate should succeed with recommendations on valid spec`(gradleVersion: String?) {
+        // Arrange
+        val projectFiles = mapOf(
+            "spec.yaml" to javaClass.classLoader.getResourceAsStream("specs/petstore-v3.0-recommend.yaml")
+        )
+
+        // recommend = true is the default
+        withProject(
+            """
+            | plugins {
+            |   id 'org.openapi.generator'
+            | }
+            |
+            | openApiValidate {
+            |   inputSpec = file("spec.yaml").absolutePath
+            | }
+        """.trimMargin(), projectFiles
+        )
+
+        // Act
+        val result = getGradleRunner(gradleVersion)
+            .withProjectDir(temp)
+            .withArguments("openApiValidate")
+            .withPluginClasspath()
+            .build()
+
+        // Assert
+        assertTrue(
+            result.output.contains("Spec is valid."),
+            "Unexpected/no message presented to the user for a valid spec."
+        )
+        assertTrue(
+            result.output.contains("Spec has issues or recommendations."),
+            "Unexpected/no recommendations message presented to the user for a valid spec."
+        )
+        assertEquals(
+            SUCCESS, result.task(":openApiValidate")?.outcome,
+            "Expected a successful run, but found ${result.task(":openApiValidate")?.outcome}"
+        )
+    }
+
+    @Test(dataProvider = "gradle_version_provider")
+    fun `openApiValidate should succeed without recommendations on valid spec`(gradleVersion: String?) {
+        // Arrange
+        val projectFiles = mapOf(
+            "spec.yaml" to javaClass.classLoader.getResourceAsStream("specs/petstore-v3.0-recommend.yaml")
+        )
+
+        withProject(
+            """
+            | plugins {
+            |   id 'org.openapi.generator'
+            | }
+            |
+            | openApiValidate {
+            |   inputSpec = file("spec.yaml").absolutePath
+            |   recommend = false
+            | }
+        """.trimMargin(), projectFiles
+        )
+
+        // Act
+        val result = getGradleRunner(gradleVersion)
+            .withProjectDir(temp)
+            .withArguments("openApiValidate")
+            .withPluginClasspath()
+            .build()
+
+        // Assert
+        assertTrue(
+            result.output.contains("Spec is valid."),
+            "Unexpected/no message presented to the user for a valid spec."
+        )
+        assertTrue(
+            result.output.contains("Spec has issues or recommendations.").not(),
+            "Unexpected/recommendations message presented to the user for a valid spec"
+        )
+        assertEquals(
+            SUCCESS, result.task(":openApiValidate")?.outcome,
+            "Expected a successful run, but found ${result.task(":openApiValidate")?.outcome}"
+        )
+    }
 }

--- a/modules/openapi-generator-gradle-plugin/src/test/resources/specs/petstore-v3.0-recommend.yaml
+++ b/modules/openapi-generator-gradle-plugin/src/test/resources/specs/petstore-v3.0-recommend.yaml
@@ -1,0 +1,114 @@
+openapi: "3.0.0"
+info:
+  version: 1.0.0
+  title: Swagger Petstore
+  license:
+    name: MIT
+servers:
+  - url: http://petstore.swagger.io/v1
+paths:
+  /pets:
+    get:
+      summary: List all pets
+      operationId: listPets
+      tags:
+        - pets
+      parameters:
+        - name: limit
+          in: query
+          description: How many items to return at one time (max 100)
+          required: false
+          schema:
+            type: integer
+            format: int32
+      requestBody: # warning `API GET/HEAD defined with request body`
+        content:
+          application/json:
+            schema:
+              $ref: "#/components/schemas/Pets"
+      responses:
+        '200':
+          description: A paged array of pets
+          headers:
+            x-next:
+              description: A link to the next page of responses
+              schema:
+                type: string
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/Pets"
+        default:
+          description: unexpected error
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/Error"
+    post:
+      summary: Create a pet
+      operationId: createPets
+      tags:
+        - pets
+      responses:
+        '201':
+          description: Null response
+        default:
+          description: unexpected error
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/Error"
+  /pets/{petId}:
+    get:
+      summary: Info for a specific pet
+      operationId: showPetById
+      tags:
+        - pets
+      parameters:
+        - name: petId
+          in: path
+          required: true
+          description: The id of the pet to retrieve
+          schema:
+            type: string
+      responses:
+        '200':
+          description: Expected response to a valid request
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/Pets"
+        default:
+          description: unexpected error
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/Error"
+components:
+  schemas:
+    Pet:
+      required:
+        - id
+        - name
+      properties:
+        id:
+          type: integer
+          format: int64
+        name:
+          type: string
+        tag:
+          type: string
+    Pets:
+      type: array
+      items:
+        $ref: "#/components/schemas/Pet"
+    Error:
+      required:
+        - code
+        - message
+      properties:
+        code:
+          type: integer
+          format: int32
+        message:
+          type: string

